### DISCUSSION TITLE
Save mp4 during Dash streaming

### DISCUSSION THREAD

**Discussion by phabloshablo** [streamlink/streamlink#4063] [Q&A] [answered]
Hi,

I need to analyze the video output so i have uploaded my dash video in my local server, and with one host e get a requesti with streamilink <link>.

I need to analyze my video output so i net do start streaming without setting "best,low ecc..." but setting option with low latency varying with my net bandwith.
How can i do it e download in local?

**Comment by mkbloke** [comment, marked as answer]
You can download from a local server by prefixing your local server URL with `dash://`, e,g,:
```
streamlink dash://https://myserver.local/path/manifest.mpd
```

Streamlink does not do automatic adjustment of the bitrate, if that is what you are wanting.

You can save output to file using the `-o` option or via `-O` and redirecting `stdout` to a file.

**Reply by phabloshablo** [reply]
thank you, yes my goal was to save videos as they appear so with resolution or quality who change based on my connection

**Reply by bastimeyer** [reply]
Streamlink doesn't support adaptive streaming with changing qualities on the fly, because it can only output a single stream, so this is not possible. Changing qualities is a feature of a player which can switch its input stream depending on the available buffer or other metrics, which is not the case with how Streamlink works and how the player reads its data.

**Reply by phabloshablo** [reply]
Ok. Thank you
